fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -274,21 +314,23 @@ public <M extends Message> CompletableFuture<Void> update(@Nullable FDBIndexable
         // delete old
         try {
             for (Tuple t : oldRecordFields.keySet()) {
-                deleteDocument(t, oldRecord.getPrimaryKey().pack());
+                if (oldRecord != null) {
```

Comment:
Is this the same as the `TODO` below where static analyzer doesn't realize that `oldRecord` can't be null at this point? Maybe it would be clearer here and there to have `Objects.requireNonNull`?

Reply:
Yes, same issue as with the `newRecord` being @Nullable. Fixed both and added comment.